astronomer/providers/google/cloud/operators/kubernetes_engine.py

```diff
@@ -116,13 +124,83 @@ def execute(self, context: Context) -> None:
                 regional=self.regional,
                 poll_interval=self.poll_interval,
                 pending_phase_timeout=self.startup_timeout_seconds,
+                logging_interval=self.logging_interval,
             ),
-            method_name="execute_complete",
+            method_name=self.trigger_reentry.__name__,
         )
 
     def execute_complete(self, context: Context, event: Dict[str, Any]) -> Any:
         """Callback for trigger once task reach terminal state"""
-        if event and event["status"] == "done":
-            self.log.info("Job completed successfully")
-        else:
-            raise AirflowException(event["description"])
+        self.trigger_reentry(context=context, event=event)
```

Comment:
this will always enter in trigger_reentry. do we need first check and if the job is not completed then only enter?

Reply:
where you are logging the pod log?

Reply:
this `trigger_reentry` function is replaced of `execute_complete` which will be get called after the trigger returns the event if even is in the failed state it will raise the error based on the failure which is called first in the `trigger_reentry` function if it is a success then only it will continue execution which will fetch the log from the `fetch_container_logs` function based on the pod which is started